### PR TITLE
Fixed core pack dependencies validation and added the use_pack_metadata flag

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36641

## Description
fixed core pack dependencies and added the use_pack_metadata flag to find dependencies command.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 1]
```diff
@@ -1205,4 +1206,27 @@ def find_dependencies(pack_name: str, id_set_path: str = '', exclude_ignored_dep
...
+
+        pack_meta_file_content = PackDependencies.get_metadata_from_pack(pack_name)
+
+        manual_dependencies = pack_meta_file_content.get('dependencies', {})
+        first_level_dependencies.update(manual_dependencies)
+
         return first_level_dependencies
+
+    @staticmethod
+    def get_metadata_from_pack(pack_name):
```
Comment (roysagi) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:1227]: add a docstring here please

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py [hunk 2]
```diff
@@ -1389,6 +1389,54 @@ def test_search_packs_by_items_names_or_ids(item_names, section_name, expected_r
     assert IsEqualFunctions.is_sets_equal(found_packs, expected_result)
 
 
+def test_find_dependencies_using_pack_metadata(mocker):
```
Comment (roysagi) [inline demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:1392]: add given-when-then docstring

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py [hunk 3]
```diff
@@ -1389,6 +1389,54 @@ def test_search_packs_by_items_names_or_ids(item_names, section_name, expected_r
...
+    }
+
+    mocker.patch('demisto_sdk.commands.find_dependencies.find_dependencies.PackDependencies.get_metadata_from_pack',
+                 return_value=mock_pack_meta_file)
+
+    first_level_dependencies = PackDependencies.update_dependencies_from_pack_metadata('', dependencies_from_id_set)
+
+    assert not first_level_dependencies.get("dependency_pack2", {}).get("mandatory") and \
+        not first_level_dependencies.get("dependency_pack3", {}).get("mandatory") and \
+        first_level_dependencies.get("dependency_pack4", {}).get("mandatory")
```
Comment (roysagi) [inline demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:1437]: split this to different asserts so we can easily identify what failed.

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 4]
```diff
@@ -1205,4 +1206,27 @@ def find_dependencies(pack_name: str, id_set_path: str = '', exclude_ignored_dep
...
             click.echo(click.style(dependency_result, bold=True))
+
+        if use_pack_metadata:
+            first_level_dependencies = PackDependencies.update_dependencies_from_pack_metadata(pack_name,
+                                                                                               first_level_dependencies)
+
+        return first_level_dependencies
+
+    @staticmethod
+    def update_dependencies_from_pack_metadata(pack_name, first_level_dependencies):
```
Comment (roysagi) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:1217]: add docstring

**Review by roysagi — APPROVED** [review]
Please add a changelog entry and add the requested docstings.

Other than that - looks good!

File: demisto_sdk/commands/find_dependencies/find_dependencies.py (hunk 4 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:1217]: done

File: demisto_sdk/commands/find_dependencies/find_dependencies.py (hunk 1 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:1227]: done

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py (hunk 2 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:1392]: done

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py (hunk 3 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:1437]: done

**Comment by yaakovpraisler** [comment]
I checked this fix on the branch we found the issue, and it seems to have been fixed.
Here is the PR that the validate failed on https://github.com/demisto/content/pull/12503.

**Pull request merged**